pandas/core/window/rolling.py

```diff
@@ -1939,7 +1937,9 @@ def aggregate(self, func, *args, **kwargs):
     def count(self):
 
         # different impl for freq counting
-        if self.is_freq_type:
+        # GH 32865. Also use custom rolling windows calculation
+        # when using a BaseIndexer subclass
+        if self.is_freq_type or isinstance(self.window, BaseIndexer):
```

Comment:
This is the crux of the issue. Because we didn't check for `BaseIndexer` subclass here, the algorithm went down the default path, and the `BaseIndexer` implementation got completely ignored as blocks were created using `_Window.create_blocks`.